pkg/controller/opensearchservice/domain/setup.go

```diff
@@ -334,3 +346,1125 @@ func GenerateObservation(obj *svcsdk.DomainStatus) svcapitypes.DomainObservation
+	}
+	if out.DomainStatus.EBSOptions != nil {
+		if obj.Spec.ForProvider.EBSOptions != nil {
+			if aws.BoolValue(obj.Spec.ForProvider.EBSOptions.EBSEnabled) != aws.BoolValue(out.DomainStatus.EBSOptions.EBSEnabled) {
+				return false, nil
+			}
+			if aws.Int64Value(obj.Spec.ForProvider.EBSOptions.IOPS) != aws.Int64Value(out.DomainStatus.EBSOptions.Iops) {
+				return false, nil
+			}
+			if aws.Int64Value(obj.Spec.ForProvider.EBSOptions.VolumeSize) != aws.Int64Value(out.DomainStatus.EBSOptions.VolumeSize) {
+				return false, nil
+			}
+			if aws.StringValue(obj.Spec.ForProvider.EBSOptions.VolumeType) != aws.StringValue(out.DomainStatus.EBSOptions.VolumeType) {
+				return false, nil
+			}
+		} else {
+			return false, nil
+		}
+	} else if obj.Spec.ForProvider.EBSOptions != nil {
+		return false, nil
+	}
+
+	if aws.StringValue(obj.Spec.ForProvider.EngineVersion) != aws.StringValue(out.DomainStatus.EngineVersion) {
+		return false, nil
+	}
+
+	if out.DomainStatus.LogPublishingOptions != nil {
+		if obj.Spec.ForProvider.LogPublishingOptions != nil {
+			if len(obj.Spec.ForProvider.LogPublishingOptions) != len(out.DomainStatus.LogPublishingOptions) {
+				return false, nil
+			}
+			for key, value := range obj.Spec.ForProvider.LogPublishingOptions {
+				if aws.StringValue(value.CloudWatchLogsLogGroupARN) != aws.StringValue(out.DomainStatus.LogPublishingOptions[key].CloudWatchLogsLogGroupArn) {
+					return false, nil
+				}
+				if aws.BoolValue(value.Enabled) != aws.BoolValue(out.DomainStatus.LogPublishingOptions[key].Enabled) {
+					return false, nil
+				}
+			}
+
+		} else {
+			return false, nil
+		}
+	} else if obj.Spec.ForProvider.LogPublishingOptions != nil {
+		return false, nil
+	}
+
+	if out.DomainStatus.NodeToNodeEncryptionOptions != nil {
+		if obj.Spec.ForProvider.NodeToNodeEncryptionOptions != nil {
+			if aws.BoolValue(obj.Spec.ForProvider.NodeToNodeEncryptionOptions.Enabled) != aws.BoolValue(out.DomainStatus.NodeToNodeEncryptionOptions.Enabled) {
+				return false, nil
+			}
+		} else {
+			return false, nil
+		}
+	} else if obj.Spec.ForProvider.NodeToNodeEncryptionOptions != nil {
+		return false, nil
+	}
+
+	if out.DomainStatus.SnapshotOptions != nil {
+		if obj.Spec.ForProvider.SnapshotOptions != nil {
+			if aws.Int64Value(out.DomainStatus.SnapshotOptions.AutomatedSnapshotStartHour) != aws.Int64Value(obj.Spec.ForProvider.SnapshotOptions.AutomatedSnapshotStartHour) {
+				return false, nil
+			}
+		}
+	} else if obj.Spec.ForProvider.SnapshotOptions != nil {
+		return false, nil
+	}
+
+	if out.DomainStatus.VPCOptions != nil {
+		if obj.Spec.ForProvider.VPCOptions != nil {
+			if len(out.DomainStatus.VPCOptions.SecurityGroupIds) != len(obj.Spec.ForProvider.VPCOptions.SecurityGroupIDRefs) {
+				return false, nil
+			}
+			for _, objValue := range obj.Spec.ForProvider.VPCOptions.SecurityGroupIDs {
+				found := false
+				for _, valueOut := range out.DomainStatus.VPCOptions.SecurityGroupIds {
+					if *objValue == *valueOut {
```

Comment:
can we use here awsclient.string ? to check for nil ?

Reply:
Updated the two places. Thanks for the hint!